onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -285,40 +285,50 @@ static inline void ComputeSoftmaxZero(std::vector<float>& values) {
 }
 
 static inline void write_scores(std::vector<float>& scores, POST_EVAL_TRANSFORM post_transform, int64_t write_index, Tensor* Z, int add_second_class) {
-  if (post_transform == POST_EVAL_TRANSFORM::PROBIT && scores.size() == 1) {
-    scores[0] = ComputeProbit(scores[0]);
-    Z->template MutableData<float>()[write_index] = scores[0];
-  } else if (scores.size() >= 2) {  //multiclass
-    if (post_transform == POST_EVAL_TRANSFORM::LOGISTIC) {
-      for (float& score : scores) {
-        score = ComputeLogistic(score);
-      }
-    } else if (post_transform == POST_EVAL_TRANSFORM::SOFTMAX) {
-      ComputeSoftmax(scores);
-    } else if (post_transform == POST_EVAL_TRANSFORM::SOFTMAX_ZERO) {
-      ComputeSoftmaxZero(scores);
+  if (scores.size() >= 2) {  //multiclass
+    switch ((unsigned char)post_transform) {
```

Comment:
>switch  [](start = 4, length = 7)

Add default: clauses to every switch and consider what should be done when it is reached. If never then assert(false)

Reply:
fixed